pkg/virt-controller/watch/vm.go

```diff
@@ -1129,7 +1129,7 @@ func (c *VMController) applyFlavorToVmi(vm *virtv1.VirtualMachine, vmi *virtv1.V
 		return nil
 	}
 
-	conflicts := c.flavorMethods.ApplyToVmi(k8sfield.NewPath("spec"), flavorProfile, &vmi.Spec)
+	conflicts := c.flavorMethods.ApplyToVmi(k8sfield.NewPath("spec"), flavorProfile, &vmi.Spec, &vmi.ObjectMeta)
```

Comment:
we could just pass the entire VMI in here to simplify things. so something like...

```conflicts := c.flavorMethods.ApplyToVmi(k8sfield.NewPath("spec"), flavorProfile, &vmi)```

Reply:
Ok I found where the problem was, but I was able to covert the object to a VMI